consensus/consensus-types/src/block.rs

```diff
@@ -485,29 +487,44 @@ where
 
 impl<T> TryFrom<network::proto::Block> for Block<T>
 where
-    T: DeserializeOwned + CanonicalDeserialize,
+    T: DeserializeOwned + CanonicalDeserialize + CanonicalSerialize,
 {
     type Error = failure::Error;
 
     fn try_from(proto: network::proto::Block) -> failure::Result<Self> {
-        let id = HashValue::from_slice(proto.id.as_ref())?;
         let timestamp_usecs = proto.timestamp_usecs;
         let epoch = proto.epoch;
         let round = proto.round;
         let quorum_cert = proto
             .quorum_cert
             .ok_or_else(|| format_err!("Missing quorum_cert"))?
             .try_into()?;
+        let author = Author::try_from(proto.author);
+        let payload = from_slice(&proto.payload);
```

Comment:
you probably should do the `?` here

Reply:
IIUC not doing so allows me to have a `None` in line 507 in case this is a genesis / nil block.